pkg/registry/core/service/strategy.go

```diff
@@ -118,7 +120,24 @@ func (svcStrategy) AllowUnconditionalUpdate() bool {
 //	    newSvc.Spec.MyFeature = nil
 //	}
 func dropServiceDisabledFields(newSvc *api.Service, oldSvc *api.Service) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.LoadBalancerIPMode) && !loadbalancerIPModeInUse(oldSvc) {
```

Comment:
These needs tests in pkg/registry/core/service/strategy_test.go and pkg/registry/core/service/storage/storage_test.go